swingset/src/main/java/com/nqadmin/swingset/formatting/SSDateField.java

```diff
@@ -85,16 +99,20 @@ public SSDateField(final int format) {
      */
     public SSDateField(final javax.swing.JFormattedTextField.AbstractFormatterFactory factory) {
         super(factory);
-        // TODO Consider setting to null vs system date.
-        setValue(new java.util.Date());
+		// TODO Consider setting to null vs system date.
+		if (!getAllowNull()) {
+			setValue(new java.util.Date());
+		}
```

Comment:
Does it matter what we do in the constructor? Nothing's bound yet, no idea what `AllowNull` is.

Reply:
Makes sense on the constructor since we don't know what getAllowNull() will give. What about cleanField() presuming we figure out how to call it when appropriate? Seems safest to set to null if allowed or you could get some unexpected dates propagating to the database. Would rather force the end user to fill explicitly.

Reply:
Even if !Allowed, could set to null. Then it shows up red.

I've got a PR ready with the background color stuff working, empty constructor, cleanField called for new record. I'll push that and you can take a look, see how it feels.